tests/catalog/test_glue.py

```diff
@@ -848,3 +848,17 @@ def test_table_exists(
     assert test_catalog.table_exists(identifier) is True
     # Act and Assert for a non-existing table
     assert test_catalog.table_exists(('non', 'exist')) is False
+
+
+@mock_aws
+def test_register_table_with_given_location(
+    _bucket_initialize: None, moto_endpoint_url: str, metadata_location: str, database_name: str, table_name: str
+) -> None:
+    catalog_name = "glue"
+    identifier = (database_name, table_name)
+    location = metadata_location
+    test_catalog = GlueCatalog(catalog_name, **{"s3.endpoint": moto_endpoint_url, "warehouse": f"s3://{BUCKET_NAME}"})
+    test_catalog.create_namespace(namespace=database_name, properties={"location": f"s3://{BUCKET_NAME}/{database_name}.db"})
+    table = test_catalog.register_table(identifier, location)
+    assert table.identifier == (catalog_name,) + identifier
+    assert test_catalog.table_exists(identifier) is True
```

Comment:
I've seen that issue come up several times in the glue context. 

Here are some related issues that might provide some hints
https://github.com/apache/iceberg-python/issues/570
https://github.com/apache/iceberg-python/issues/515
https://github.com/apache/iceberg-python/issues/692

Reply:
Seems like an issue on my local, it is running fine on git actions.

Reply:
I also got the same error when running the integration test. The integration test will try to purge the remaining tables created during the test as a final step:
https://github.com/apache/iceberg-python/blob/62b527e74cff328fde5faa848519c56748a244c1/tests/conftest.py#L1919-L1926

Since `test_register_table` use a mocked metadata file, the manifest files recorded in the metadata file does not exist. Hence, we encounter error here since the `clean_up` try to delete a manifest file that never existed.

Reply:
thanks a lot, Honah for the detailed overview, it all makes sense now.